libcontainer/cgroups/fs/cpuacct.go

```diff
@@ -120,3 +135,43 @@ func getPercpuUsage(path string) ([]uint64, error) {
 	}
 	return percpuUsage, nil
 }
+
+func getPercpuUsageInModes(path string) ([]uint64, []uint64, error) {
+	percpuUsageKernelMode := []uint64{}
+	percpuUsageUserMode := []uint64{}
+
+	file, err := os.Open(filepath.Join(path, cgroupCpuacctUsageAll))
+	if err != nil {
+		return []uint64{}, []uint64{}, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	scanner.Scan() //skipping header line
+
+	for scanner.Scan() {
+		lineFields := strings.Fields(scanner.Text())
```

Comment:
using `strings.SplitN` is usually 2x faster, and you can limit the number of columns (so it won't keep splitting in case of bad input)